src/bosh-director/lib/bosh/director/config_server/variables_interpolator.rb

```diff
@@ -4,13 +4,14 @@ class VariablesInterpolator
     def initialize
       @logger = Bosh::Director::Config.logger
       @config_server_client = ClientFactory.create(@logger).create_client
+      @cache_by_job_name = {}
```

Comment:
Every BOSH task will have its own cache, which is used during the execution of that task and the cache is shared between all instances of that task.  The cache (the class `VariablesInterpolator`) gets garbage collected after the task is finished. So we don't have any long living cache in memory, which can cause creds rotation issues. Is my understanding correct?